### PR TITLE
New package r-tidyverse and r-tidyselect

### PR DESCRIPTION
**Pull request by fanne-stat**
- New package r-tidyverse and r-tidyselect
- New version and dependencies added for some packages

### CODE REVIEW

File: var/spack/repos/builtin/packages/r-tidyr/package.py [hunk 1]
```diff
@@ -31,14 +31,18 @@ class RTidyr(RPackage):
...
     homepage = "https://github.com/hadley/tidyr"
-    url      = "https://cran.r-project.org/src/contrib/tidyr_0.5.1.tar.gz"
+    url      = "https://cran.r-project.org/src/contrib/tidyr_0.7.2.tar.gz"
     list_url = "https://cran.r-project.org/src/contrib/Archive/tidyr"
 
+    version('0.7.2', '42d723bf04c5c1c59e27a8be14f3a6b6')
     version('0.5.1', '3cadc869510c054ed93d374ab44120bd')
 
     depends_on('r-tibble', type=('build', 'run'))
     depends_on('r-dplyr', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-tidyr/package.py]: Can you add the version constraints:
```python
depends_on('r-dplyr@0.7.0:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/r-tidyr/package.py [hunk 2]
```diff
@@ -31,14 +31,18 @@ class RTidyr(RPackage):
...
     depends_on('r-stringi', type=('build', 'run'))
     depends_on('r-lazyeval', type=('build', 'run'))
     depends_on('r-magrittr', type=('build', 'run'))
+    depends_on('r-glue', type=('build', 'run'))
     depends_on('r-rcpp', type=('build', 'run'))
+    depends_on('r-purrr', type=('build', 'run'))
+    depends_on('r-tidyselect', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-tidyr/package.py]: Missing a dependency on `r-rlang`.

File: var/spack/repos/builtin/packages/r-tidyr/package.py [hunk 3]
```diff
@@ -31,14 +31,18 @@ class RTidyr(RPackage):
...
+    url      = "https://cran.r-project.org/src/contrib/tidyr_0.7.2.tar.gz"
     list_url = "https://cran.r-project.org/src/contrib/Archive/tidyr"
 
+    version('0.7.2', '42d723bf04c5c1c59e27a8be14f3a6b6')
     version('0.5.1', '3cadc869510c054ed93d374ab44120bd')
 
     depends_on('r-tibble', type=('build', 'run'))
     depends_on('r-dplyr', type=('build', 'run'))
     depends_on('r-stringi', type=('build', 'run'))
     depends_on('r-lazyeval', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-tidyr/package.py]: `r-lazyeval` isn't a dependency as far as I can tell.

**Comment by baberlevi** [comment]
@fanne-stat  - reminder that you still need to add the missing r-rlang dependency to r-tidyr

**Comment by adamjstewart** [comment]
Ping @fanne-stat

File: var/spack/repos/builtin/packages/r-tidyr/package.py (hunk 2 above)
Comment (fanne-stat) [reply var/spack/repos/builtin/packages/r-tidyr/package.py]: r-rlang is added to dependency now. @adamjstewart

File: var/spack/repos/builtin/packages/r-tidyverse/package.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+##############################################################################
+from spack import *
+
+
+class RTidyverse(RPackage):
+    """The 'tidyverse' is a set of packages that work in harmony because they
+       share common data representations and 'API' design. This package is
+       designed to make it easy to install and load multiple 'tidyverse'
+       packages in a single step. Learn more about the 'tidyverse' at
+       <https://tidyverse.org>."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-tidyverse/package.py]: This URL might be causing the documentation tests to fail. You can probably remove the last sentence, or even the last 2 sentences.

File: var/spack/repos/builtin/packages/r-tidyverse/package.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    depends_on('r-readxl@1.0.0:', type=('build', 'run'))
+    depends_on('r-reprex@0.1.1:', type=('build', 'run'))
+    depends_on('r-rlang@0.1.4:', type=('build', 'run'))
+    depends_on('r-rstudioapi@0.7:', type=('build', 'run'))
+    depends_on('r-rvest@0.3.2:', type=('build', 'run'))
+    depends_on('r-stringr@1.2.0:', type=('build', 'run'))
+    depends_on('r-tibble@1.3.4:', type=('build', 'run'))
+    depends_on('r-tidyr@0.7.2:', type=('build', 'run'))
+    depends_on('r-xml2@1.1.1:', type=('build', 'run'))
+    depens_on('r-rlang@0.1.4:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-tidyverse/package.py]: Typo: depens_on

**Pull request merged**